### PR TITLE
Collapses Identical catch sections in a single try statement.

### PR DESCRIPTION
**Pull request by arturobernalg**

### CODE REVIEW

File: maven-scm-providers/maven-scm-provider-clearcase/src/main/java/org/apache/maven/scm/provider/clearcase/util/ClearCaseUtil.java [hunk 1]
```diff
@@ -82,16 +82,10 @@ public static Settings readSettings()
...
+            } catch ( IOException e )
             {
                 // nop
-            }
-            catch ( IOException e )
-            {
-                // nop
-            }
-            catch ( XmlPullParserException e )
+            } catch ( XmlPullParserException e )
```
Comment (michael-o) [inline maven-scm-providers/maven-scm-provider-clearcase/src/main/java/org/apache/maven/scm/provider/clearcase/util/ClearCaseUtil.java:88]: Please explain

File: maven-scm-providers/maven-scm-provider-perforce/src/main/java/org/apache/maven/scm/provider/perforce/PerforceScmProvider.java [hunk 2]
```diff
@@ -433,16 +433,11 @@ private static String generateDefaultClientspecName( ScmLogger logger, PerforceS
...
-        catch ( UnknownHostException e )
+        } catch ( IOException e )
         {
             // Should never happen
             throw new RuntimeException( e );
         }
-        catch ( IOException e )
-        {
-            throw new RuntimeException( e );
-        }
```
Comment (michael-o) [inline maven-scm-providers/maven-scm-provider-perforce/src/main/java/org/apache/maven/scm/provider/perforce/PerforceScmProvider.java:445]: Same here.

File: maven-scm-providers/maven-scm-provider-vss/src/main/java/org/apache/maven/scm/provider/vss/commands/VssCommandLineUtils.java [hunk 3]
```diff
@@ -138,16 +138,10 @@ public static Settings readSettings()
...
+            } catch ( IOException e )
             {
                 // nop
-            }
-            catch ( IOException e )
-            {
-                // nop
-            }
-            catch ( XmlPullParserException e )
+            } catch ( XmlPullParserException e )
```
Comment (michael-o) [inline maven-scm-providers/maven-scm-provider-vss/src/main/java/org/apache/maven/scm/provider/vss/commands/VssCommandLineUtils.java:144]: Same here.

File: maven-scm-providers/maven-scm-providers-cvs/maven-scm-provider-cvs-commons/src/main/java/org/apache/maven/scm/provider/cvslib/util/CvsUtil.java [hunk 4]
```diff
@@ -67,16 +67,10 @@ public static Settings readSettings()
...
-            {
-                // skip error 
-            }
-            catch ( IOException e )
+            } catch ( IOException e )
             {
                 // skip error
-            }
-            catch ( XmlPullParserException e )
+            } catch ( XmlPullParserException e )
```
Comment (michael-o) [inline maven-scm-providers/maven-scm-providers-cvs/maven-scm-provider-cvs-commons/src/main/java/org/apache/maven/scm/provider/cvslib/util/CvsUtil.java:73]: Same here.

File: maven-scm-providers/maven-scm-providers-git/maven-scm-provider-git-commons/src/main/java/org/apache/maven/scm/provider/git/util/GitUtil.java [hunk 5]
```diff
@@ -66,16 +66,10 @@ public static Settings readSettings()
...
+            } catch ( IOException e )
             {
                 //Nothing to do
-            }
-            catch ( IOException e )
-            {
-                //Nothing to do
-            }
-            catch ( XmlPullParserException e )
+            } catch ( XmlPullParserException e )
```
Comment (michael-o) [inline maven-scm-providers/maven-scm-providers-git/maven-scm-provider-git-commons/src/main/java/org/apache/maven/scm/provider/git/util/GitUtil.java:72]: Same here.

File: maven-scm-providers/maven-scm-providers-svn/maven-scm-provider-svn-commons/src/main/java/org/apache/maven/scm/provider/svn/util/SvnUtil.java [hunk 6]
```diff
@@ -65,16 +65,10 @@ public static Settings readSettings()
...
+            } catch ( IOException e )
             {
                 //Nothing to do
-            }
-            catch ( IOException e )
-            {
-                //Nothing to do
-            }
-            catch ( XmlPullParserException e )
+            } catch ( XmlPullParserException e )
```
Comment (michael-o) [inline maven-scm-providers/maven-scm-providers-svn/maven-scm-provider-svn-commons/src/main/java/org/apache/maven/scm/provider/svn/util/SvnUtil.java:71]: Same here.

**Review by michael-o — CHANGES_REQUESTED** [review]

**Comment by arturobernalg** [comment]
HI @michael-o 
'catch' sections are identical to 'FileNotFoundException'. Maybe would be better a custom message if any IO error comes?
TY

**Review by michael-o — CHANGES_REQUESTED** [review]
Yes, you are right. Makes sense. Retain formatting please: catch starts on a new line.

**Comment by arturobernalg** [comment]
> Yes, you are right. Makes sense. Retain formatting please: catch starts on a new line.

HI @michael-o 
TY. I can´t  find where need to fix formatting.
TY

File: maven-scm-providers/maven-scm-provider-clearcase/src/main/java/org/apache/maven/scm/provider/clearcase/util/ClearCaseUtil.java [hunk 7]
```diff
@@ -82,16 +82,10 @@ public static Settings readSettings()
             try
             {
                 return reader.read( ReaderFactory.newXmlReader( settingsFile ) );
-            }
-            catch ( FileNotFoundException e )
+            } catch ( IOException e )
```
Comment (michael-o) [inline maven-scm-providers/maven-scm-provider-clearcase/src/main/java/org/apache/maven/scm/provider/clearcase/util/ClearCaseUtil.java:85]: Here

File: maven-scm-providers/maven-scm-provider-clearcase/src/main/java/org/apache/maven/scm/provider/clearcase/util/ClearCaseUtil.java (hunk 1 above)
Comment (michael-o) [inline maven-scm-providers/maven-scm-provider-clearcase/src/main/java/org/apache/maven/scm/provider/clearcase/util/ClearCaseUtil.java:88]: Here

File: maven-scm-providers/maven-scm-provider-perforce/src/main/java/org/apache/maven/scm/provider/perforce/PerforceScmProvider.java [hunk 8]
```diff
@@ -433,16 +433,11 @@ private static String generateDefaultClientspecName( ScmLogger logger, PerforceS
             hostname = InetAddress.getLocalHost().getHostName();
             // [SCM-370][SCM-351] client specs cannot contain forward slashes, spaces and ~; "-" is okay
             path = workDir.getCanonicalPath().replaceAll( "[/ ~]", "-" );
-        }
-        catch ( UnknownHostException e )
+        } catch ( IOException e )
```
Comment (michael-o) [inline maven-scm-providers/maven-scm-provider-perforce/src/main/java/org/apache/maven/scm/provider/perforce/PerforceScmProvider.java:436]: Here

File: maven-scm-providers/maven-scm-provider-perforce/src/main/java/org/apache/maven/scm/provider/perforce/command/PerforceInfoCommand.java [hunk 9]
```diff
@@ -153,15 +153,10 @@ protected ScmResult executeCommand( ScmProviderRepository repo, ScmFileSet scmFi
...
+        catch ( CommandLineException | IOException e )
         {
             throw new ScmException( e.getLocalizedMessage() );
-        }
-        catch ( IOException e )
-        {
-            throw new ScmException( e.getLocalizedMessage() );
-        }
-        finally
+        } finally
```
Comment (michael-o) [inline maven-scm-providers/maven-scm-provider-perforce/src/main/java/org/apache/maven/scm/provider/perforce/command/PerforceInfoCommand.java:159]: Here

File: maven-scm-providers/maven-scm-provider-perforce/src/main/java/org/apache/maven/scm/provider/perforce/command/PerforceWhereCommand.java [hunk 10]
```diff
@@ -149,23 +149,14 @@ public String getDepotLocation( String filepath )
...
-        catch ( IOException e )
-        {
-            if ( logger.isErrorEnabled() )
-            {
-                logger.error( e );
-            }
-            throw new RuntimeException( e.getLocalizedMessage() );
-        }
-        finally
+        } finally
```
Comment (michael-o) [inline maven-scm-providers/maven-scm-provider-perforce/src/main/java/org/apache/maven/scm/provider/perforce/command/PerforceWhereCommand.java:159]: Here

File: maven-scm-providers/maven-scm-provider-starteam/src/main/java/org/apache/maven/scm/provider/starteam/util/StarteamUtil.java [hunk 11]
```diff
@@ -65,16 +65,10 @@ public static Settings readSettings()
             try
             {
                 return reader.read( ReaderFactory.newXmlReader( settingsFile ) );
-            }
-            catch ( FileNotFoundException e )
+            } catch ( IOException e )
```
Comment (michael-o) [inline maven-scm-providers/maven-scm-provider-starteam/src/main/java/org/apache/maven/scm/provider/starteam/util/StarteamUtil.java:68]: Here

File: maven-scm-providers/maven-scm-provider-starteam/src/main/java/org/apache/maven/scm/provider/starteam/util/StarteamUtil.java [hunk 12]
```diff
@@ -65,16 +65,10 @@ public static Settings readSettings()
...
             {
                 // nop
-            }
-            catch ( IOException e )
-            {
-                // nop
-            }
-            catch ( XmlPullParserException e )
+            } catch ( XmlPullParserException e )
```
Comment (michael-o) [inline maven-scm-providers/maven-scm-provider-starteam/src/main/java/org/apache/maven/scm/provider/starteam/util/StarteamUtil.java:71]: Here

File: maven-scm-providers/maven-scm-provider-vss/src/main/java/org/apache/maven/scm/provider/vss/commands/VssCommandLineUtils.java [hunk 13]
```diff
@@ -138,16 +138,10 @@ public static Settings readSettings()
             try
             {
                 settings = reader.read( ReaderFactory.newXmlReader( settingsFile ) );
-            }
-            catch ( FileNotFoundException e )
+            } catch ( IOException e )
```
Comment (michael-o) [inline maven-scm-providers/maven-scm-provider-vss/src/main/java/org/apache/maven/scm/provider/vss/commands/VssCommandLineUtils.java:141]: Here

File: maven-scm-providers/maven-scm-provider-vss/src/main/java/org/apache/maven/scm/provider/vss/commands/VssCommandLineUtils.java (hunk 3 above)
Comment (michael-o) [inline maven-scm-providers/maven-scm-provider-vss/src/main/java/org/apache/maven/scm/provider/vss/commands/VssCommandLineUtils.java:144]: Here

File: maven-scm-providers/maven-scm-providers-cvs/maven-scm-provider-cvs-commons/src/main/java/org/apache/maven/scm/provider/cvslib/util/CvsUtil.java [hunk 14]
```diff
@@ -67,16 +67,10 @@ public static Settings readSettings()
             try
             {
                 return reader.read( ReaderFactory.newXmlReader( settingsFile ) );
-            }
-            catch ( FileNotFoundException e )
-            {
-                // skip error 
-            }
-            catch ( IOException e )
+            } catch ( IOException e )
```
Comment (michael-o) [inline maven-scm-providers/maven-scm-providers-cvs/maven-scm-provider-cvs-commons/src/main/java/org/apache/maven/scm/provider/cvslib/util/CvsUtil.java:70]: Here

File: maven-scm-providers/maven-scm-providers-cvs/maven-scm-provider-cvs-commons/src/main/java/org/apache/maven/scm/provider/cvslib/util/CvsUtil.java (hunk 4 above)
Comment (michael-o) [inline maven-scm-providers/maven-scm-providers-cvs/maven-scm-provider-cvs-commons/src/main/java/org/apache/maven/scm/provider/cvslib/util/CvsUtil.java:73]: Here

File: maven-scm-providers/maven-scm-providers-git/maven-scm-provider-git-commons/src/main/java/org/apache/maven/scm/provider/git/util/GitUtil.java [hunk 15]
```diff
@@ -66,16 +66,10 @@ public static Settings readSettings()
             try
             {
                 return reader.read( ReaderFactory.newXmlReader( settingsFile ) );
-            }
-            catch ( FileNotFoundException e )
+            } catch ( IOException e )
```
Comment (michael-o) [inline maven-scm-providers/maven-scm-providers-git/maven-scm-provider-git-commons/src/main/java/org/apache/maven/scm/provider/git/util/GitUtil.java:69]: Here

File: maven-scm-providers/maven-scm-providers-git/maven-scm-provider-git-commons/src/main/java/org/apache/maven/scm/provider/git/util/GitUtil.java (hunk 5 above)
Comment (michael-o) [inline maven-scm-providers/maven-scm-providers-git/maven-scm-provider-git-commons/src/main/java/org/apache/maven/scm/provider/git/util/GitUtil.java:72]: Here

File: maven-scm-providers/maven-scm-providers-svn/maven-scm-provider-svn-commons/src/main/java/org/apache/maven/scm/provider/svn/util/SvnUtil.java [hunk 16]
```diff
@@ -65,16 +65,10 @@ public static Settings readSettings()
             try
             {
                 return reader.read( ReaderFactory.newXmlReader( settingsFile ) );
-            }
-            catch ( FileNotFoundException e )
+            } catch ( IOException e )
```
Comment (michael-o) [inline maven-scm-providers/maven-scm-providers-svn/maven-scm-provider-svn-commons/src/main/java/org/apache/maven/scm/provider/svn/util/SvnUtil.java:68]: Here

File: maven-scm-providers/maven-scm-providers-svn/maven-scm-provider-svn-commons/src/main/java/org/apache/maven/scm/provider/svn/util/SvnUtil.java (hunk 6 above)
Comment (michael-o) [inline maven-scm-providers/maven-scm-providers-svn/maven-scm-provider-svn-commons/src/main/java/org/apache/maven/scm/provider/svn/util/SvnUtil.java:71]: Here

**Review by michael-o — CHANGES_REQUESTED** [review]
Marked all formatting issues

**Comment by arturobernalg** [comment]
HI @michael-o 
now i see the error. Fixed.
TY

**Review by michael-o — APPROVED** [review]

**Pull request closed without merging**